dns-controller/pkg/dns/dnscontroller.go

```diff
@@ -363,6 +372,44 @@ func (o *dnsOp) findZone(fqdn string) dnsprovider.Zone {
 	}
 }
 
+func (o *dnsOp) getChangeset(zone dnsprovider.Zone) (dnsprovider.ResourceRecordChangeset, error) {
+	key := zone.Name() + "::" + zone.ID()
+	changeset := o.changesets[key]
+	if changeset == nil {
+		rrsProvider, ok := zone.ResourceRecordSets()
+		if !ok {
+			return nil, fmt.Errorf("zone does not support resource records %q", zone.Name())
+		}
+		changeset = rrsProvider.StartChangeset()
+		o.changesets[key] = changeset
+	}
+
+	return changeset, nil
+}
+
+// listRecords is a wrapper around listing records, but will cache the results for the duration of the dnsOp
+func (o *dnsOp) listRecords(zone dnsprovider.Zone) ([]dnsprovider.ResourceRecordSet, error) {
+	key := zone.Name() + "::" + zone.ID()
+
+	rrs := o.recordsCache[key]
+	if rrs == nil {
+		rrsProvider, ok := zone.ResourceRecordSets()
+		if !ok {
+			return nil, fmt.Errorf("zone does not support resource records %q", zone.Name())
+		}
+
+		glog.V(2).Infof("Querying all dnsprovider records for zone %q", zone.Name())
```

Comment:
Well hopefully not a lot, otherwise we will have rate limit problems.  Are you suggesting to go all the way to Infof?